File: extractor.py
import gemeni
import parse
import json
import datetime
def get_tasks(syllabus):
    response = gemeni.GetResponse(syllabus)
    try:
        decoded_response = json.loads(response)
    except:
        return "Couldn't generate tasks."
    tasks = [parse.MakeTask(unparsed_task) for unparsed_task in decoded_response]
    parsed_tasks = []
    for task in tasks:
        title = task["title"]
        date = datetime.datetime.strptime(task["due"][:-1],"%Y-%m-%dT%H:%M:%S").strftime("%m-%d")
        parsed_tasks.append(f"{date}: {title}")
    return tasks,parsed_tasks

File: extractor.js
const { GetResponse } = require("./gemeni.js");
const { MakeTask } = require("./parse.js");

async function get_tasks(key, syllabus) {
    try {
        const decoded_response = await GetResponse(key, syllabus);

        if (!Array.isArray(decoded_response)) {
            return [[], []];
        }

        const tasks = decoded_response.map(MakeTask);
        const parsed_tasks = tasks.map(task => `${task.due}: ${task.title}`);

        return [tasks, parsed_tasks];
    } catch (error) {
        console.error("Error handling Gemini response:", error);
        return [[], []];
    }
}

module.exports = { get_tasks };
